File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/github/helpers/DockerComposeClient.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl.github.helpers;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GithubClient;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Constants;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Language;

import static org.assertj.core.api.Assertions.assertThat;

public class DockerComposeClient {
    private final GithubClient client;
    private final String repositoryPath;

    public DockerComposeClient(GithubClient client) {
        this.client = client;
        this.repositoryPath = client.getRepositoryPath();
    }

    public void verifyDockerComposeImage(Language systemLanguage, Language systemTestLanguage) {
        var dockerComposePath = String.format("system-test-%s/docker-compose.yml", systemTestLanguage.getValue());

        for(var l : Language.getAll()) {
            var monolithDockerImageName = String.format(Constants.MONOLITH_DOCKER_IMAGE_NAME_FORMAT, repositoryPath, l.getValue());
            if(l.equals(systemLanguage)) {
                verifyDockerComposeContainsImage(dockerComposePath, monolithDockerImageName);
            } else {
                verifyDockerComposeDoesNotContainImage(dockerComposePath, monolithDockerImageName);
            }
        }
    }

    private void verifyDockerComposeContainsImage(String dockerComposePath, String image) {
        var dockerComposeContent = client.getFileContent(dockerComposePath);
        assertThat(dockerComposeContent)
                .as("Docker Compose should contain image: " + image)
                .contains(image);
    }

    private void verifyDockerComposeDoesNotContainImage(String dockerComposePath, String image) {
        var dockerComposeContent = client.getFileContent(dockerComposePath);
        assertThat(dockerComposeContent)
                .as("Docker Compose should not contain image: " + image)
                .doesNotContain(image);
    }
}

File: console/tests/Dsl/GitHub/Helpers/DockerComposeClient.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util;
using Shouldly;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl.GitHub.Helpers
{
    public class DockerComposeClient
    {
        private readonly GithubClient _client;
        private readonly string _repositoryPath;

        public DockerComposeClient(GithubClient client)
        {
            _client = client;
            _repositoryPath = client.GetRepositoryPath();
        }

        public void VerifyDockerComposeImage(Language systemLanguage, Language systemTestLanguage)
        {
            var dockerComposePath = $"system-test-{systemTestLanguage.GetValue()}/docker-compose.yml";

            foreach (var l in LanguageExtensions.GetAll())
            {
                var monolithDockerImageName = string.Format(Constants.MonolithDockerImageNameFormat, _repositoryPath, l.GetValue());
                if (l.Equals(systemLanguage))
                {
                    VerifyDockerComposeContainsImage(dockerComposePath, monolithDockerImageName);
                }
                else
                {
                    VerifyDockerComposeDoesNotContainImage(dockerComposePath, monolithDockerImageName);
                }
            }
        }

        private void VerifyDockerComposeContainsImage(string dockerComposePath, string image)
        {
            var dockerComposeContent = _client.GetFileContent(dockerComposePath);
            dockerComposeContent.ShouldContain(image, Case.Insensitive, $"Docker Compose should contain image: {image}");
        }

        private void VerifyDockerComposeDoesNotContainImage(string dockerComposePath, string image)
        {
            var dockerComposeContent = _client.GetFileContent(dockerComposePath);
            dockerComposeContent.ShouldNotContain(image, Case.Insensitive, $"Docker Compose should not contain image: {image}");
        }
    }
}